cpu/atmega_common/periph/rtt.c

```diff
@@ -18,19 +18,34 @@
  *              oscillator.
  *
  * @author      Matthew Blue <matthew.blue.neuro@gmail.com>
+ * @author      Alexander Chudov <chudov@gmail.com>
  *
+ * For all atmega except rfr2
  * In order to safely sleep when using the RTT:
  * 1. Disable interrupts
  * 2. Write to one of the asynch registers (e.g. TCCR2A)
  * 3. Wait for ASSR register's busy flags to clear
  * 4. Re-enable interrupts
  * 5. Sleep before interrupt re-enable takes effect
  *
+ * For MCUs with MAC symbol counter (atmega*rfr2):
+ * The MAC symbol counter is automatically switch to TOSC1 32.768kHz clock
+ * when transceiver or CPU is going to sleep. The MAC symbol counter is
+ * sourced by 62.500 kHz derived from 32.768kHz TOSC1 or 16 MHz system clock
+ * For current implementation symbol counter is always use 32.768kHz TOSC1 clock
+ * For alarms the SCOCR2 register is used.
```

Comment:
I wil take the proposed text, except place where it is little bit incorrect:  according to the datasheet "The symbol counter is a 32 bit counter which can be sourced by a 62.5 kHz clock, derived from the 16 MHz system clock or from the RTC (32.768 kHz)."

Reply:
You're right. I got confused while reading the data sheet. Thanks for pointing this out!

Reply:
Hmm, on another page the data sheet says "The counter time base can be derived from the 16 MHz crystal or the RTC (32.768 kHz crystal on TOSC) during operation."

Maybe the "62.5 kHz clock, derived from the 16 MHz system clock" does not mean it can be either a 62.5 kHz clock or the be derived from the system clock. Maybe it means that the 62.5 kHz clock is derived from the system clock by dividing the 16 MHz by 256?

Reply:
Actually in case of deriving from 16MHz clock it is possible to use another divider. See 10.11.35 SCCR1 section where it is possible to set another divider. This is nice flexibility but it is unclear for me how to use it in real 802.15.4 applications.
Even worse, the counter can be clocked from PG2 pin. I'm not sure it make sense to paste datasheet text to sources...

Reply:
> I'm not sure it make sense to paste datasheet text to sources...

No, only the takeaway needed for understanding the code and the design decisions are needed. To me, here it is: There are multiple options for the clock source. But only the 32.768kHz RTT clock source is available when either the CPU or the transceiver is not active. So therefore, the RTC is chosen as clock source here.

Reply:
Greate feature of the symbol counter that in case of deep sleep it is automatically switched from XTAL to TOSC1 and back. But that leads to some delays for oscillator stabilization and some counter resync that are not yet clear for me. This is why 32K is always used :)

Reply:
The comment you just made explains it quite good ;-)

The counter resync would also increase ROM and wake up time. It would increase the complexity of the code. To me, it is just not worth the effort.

Let me try to extend your comment above:

The ATmegaXXXRFA1/RFR2 provide multiple clock sources. But only the 32.768kHz RTT (TOSC1) is available when CPU or transceiver or not active. Thus, it is used automatically as a fallback during sleep modes regardless of configured clock source. To avoid the calculations needed to compensate for a change of the clock frequency during sleep, this implementation just uses TOSC1(32.768 kHz) at any point in time.